pyfive/dataobjects.py

```diff
@@ -250,6 +250,12 @@ def _parse_attribute_msg(self, buffer, offset):
 
     def _attr_value(self, dtype, buf, count, offset):
         """ Retrieve an HDF5 attribute value from a buffer. """
+
+        # first handle ENUMERATION, we just extract the dtype
+        if isinstance(dtype, tuple):
```

Comment:
Extracting this in that early stage here works just fine. This then runs the normal path and seems to fix the specific problem for enum attributes.